### PR TITLE
Bootstrap with EVM by default

### PR DESCRIPTION
**Pull request by janezpodhostnik**
closes: https://github.com/onflow/flow-go/issues/5491

Should be ported back to master as well

### CODE REVIEW

**Review by turbolent — APPROVED** [review]

File: fvm/bootstrap.go [hunk 1]
```diff
@@ -410,14 +447,19 @@ func (b *bootstrapExecutor) Execute() error {
...
+		return err
+	}
+
 	b.registerNodes(service, fungibleToken, flowToken)
 
 	// set the list of nodes which are allowed to stake in this network
 	b.setStakingAllowlist(service, b.identities.NodeIDs())
 
-	// sets up the EVM environment
-	b.setupEVM(service, fungibleToken, flowToken)
```
Comment (jordanschalm) [inline fvm/bootstrap.go:419]: Ah, so this is why the indices were off last time. We set up the EVM after creating the node accounts 💡

File: fvm/systemcontracts/system_contracts.go [hunk 2]
```diff
@@ -65,6 +65,8 @@ const (
 	FlowTokenAccountIndex     = 3
 	FlowFeesAccountIndex      = 4
 	EVMStorageAccountIndex    = 5
+
+	LastSystemAccountIndex = EVMStorageAccountIndex
```
Comment (jordanschalm) [inline fvm/systemcontracts/system_contracts.go:69]: I suggest adding additional documentation here.

These are the account indexes of system contracts **as deployed by the default bootstrapping**.

However, this package also contains information about chains which were not deployed by the default bootstrapping, like Mainnet and Testnet. For those chains, I don't think these indices will point to the right contracts.

**Review by jordanschalm — APPROVED** [review]

File: cmd/bootstrap/utils/key_generation_test.go [hunk 3]
```diff
@@ -62,7 +64,7 @@ func TestWriteMachineAccountFiles(t *testing.T) {
 	expected := make(map[string]bootstrap.NodeMachineAccountInfo)
 	for i, node := range nodes {
 		// See comments in WriteMachineAccountFiles for why addresses take this form
-		addr, err := chain.AddressAtIndex(uint64(6 + i*2))
+		addr, err := chain.AddressAtIndex(uint64(systemcontracts.LastSystemAccountIndex + (i+1)*2))
```
Comment (janezpodhostnik) [inline cmd/bootstrap/utils/key_generation_test.go:67]: @jordanschalm can you confirm this logic is correct?

File: fvm/bootstrap.go (hunk 1 above)
Comment (janezpodhostnik) [reply fvm/bootstrap.go:419]: Yes! I did not see that registerNodes creates new accounts.

File: cmd/bootstrap/utils/key_generation_test.go (hunk 3 above)
Comment (jordanschalm) [reply cmd/bootstrap/utils/key_generation_test.go:67]: It looks correct to me 👍

File: fvm/bootstrap.go (hunk 1 above)
Comment (turbolent) [reply fvm/bootstrap.go:419]: Maybe leave a warning in the code to make it more obvious that `registerNodes` creates accounts (maybe even improve naming) and maybe that contracts should be deployed before it.

**Review by turbolent — APPROVED** [review]
👍

**Comment by turbolent** [comment]
@janezpodhostnik The merge of this PR broke the tests on the target branch, see https://github.com/onflow/flow-go/actions/runs/8116077900/job/22185298828. Can you please have a look?

**Pull request merged**